Jellyfin.Api/Controllers/ImageController.cs

```diff
@@ -0,0 +1,1301 @@
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [SuppressMessage("Microsoft.Performance", "CA1801:ReviewUnusedParameters", MessageId = "index", Justification = "Imported from ServiceStack")]
+        public async Task<ActionResult> SetItemImage(
+            [FromRoute] Guid itemId,
+            [FromRoute] ImageType imageType,
+            [FromRoute] int? imageIndex = null)
+        {
+            var item = _libraryManager.GetItemById(itemId);
+            if (item == null)
+            {
+                return NotFound();
+            }
+
+            // Handle image/png; charset=utf-8
+            var mimeType = Request.ContentType.Split(';').FirstOrDefault();
+            await _providerManager.SaveImage(item, Request.Body, mimeType, imageType, null, CancellationToken.None).ConfigureAwait(false);
+            item.UpdateToRepository(ItemUpdateType.ImageUpdate, CancellationToken.None);
+
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Updates the index for an item image.
+        /// </summary>
+        /// <param name="itemId">Item id.</param>
+        /// <param name="imageType">Image type.</param>
+        /// <param name="imageIndex">Old image index.</param>
+        /// <param name="newIndex">New image index.</param>
+        /// <response code="204">Image index updated.</response>
+        /// <response code="404">Item not found.</response>
+        /// <returns>A <see cref="NoContentResult"/> on success, or a <see cref="NotFoundResult"/> if item not found.</returns>
+        [HttpPost("Items/{itemId}/Images/{imageType}/{imageIndex}/Index")]
+        [Authorize(Policy = Policies.RequiresElevation)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public ActionResult UpdateItemImageIndex(
+            [FromRoute] Guid itemId,
+            [FromRoute] ImageType imageType,
+            [FromRoute] int imageIndex,
+            [FromQuery] int newIndex)
+        {
+            var item = _libraryManager.GetItemById(itemId);
+            if (item == null)
+            {
+                return NotFound();
+            }
+
+            item.SwapImages(imageType, imageIndex, newIndex);
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Get item image infos.
+        /// </summary>
+        /// <param name="itemId">Item id.</param>
+        /// <response code="200">Item images returned.</response>
+        /// <response code="404">Item not found.</response>
+        /// <returns>The list of image infos on success, or <see cref="NotFoundResult"/> if item not found.</returns>
+        [HttpGet("Items/{itemId}/Images")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public ActionResult<IEnumerable<ImageInfo>> GetItemImageInfos([FromRoute] Guid itemId)
```

Comment:
I have lost the overview of the changes, so I am not sure if there's a default policy in place (imo there should never be one).

Seems like this controller is missing the default auth though.

Reply:
The API doesn't require authorization in general. We either apply it for the whole controller or for each endpoint. This depends, whether every endpoint in one controller requires authorization or not.

So yes, the two image endpoints you pointed out, are missing the authorization attribute.